examples/nidaqmx_analog_input/measurement.py

```diff
@@ -15,34 +25,67 @@
     version="0.1.0.0",
     ui_file_paths=[service_directory / "NIDAQmxAnalogInput.measui"],
 )
+service_options = ServiceOptions()
 
 
 @measurement_service.register_measurement
-@measurement_service.configuration("physical_channel", nims.DataType.String, "Dev1/ai0")
+@measurement_service.configuration(
+    "pin_name",
+    nims.DataType.Pin,
+    "Pin1",
+    instrument_type=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+)
 @measurement_service.configuration("sample_rate", nims.DataType.Double, 1000.0)
 @measurement_service.configuration("number_of_samples", nims.DataType.UInt64, 100)
 @measurement_service.output("acquired_samples", nims.DataType.DoubleArray1D)
-def measure(physical_channel, sample_rate, number_of_samples):
+def measure(pin_name, sample_rate, number_of_samples):
     """Perform a finite analog input measurement with NI-DAQmx."""
     logging.info(
-        "Executing measurement: physical_channel=%s sample_rate=%g number_of_samples=%d",
-        physical_channel,
+        "Executing measurement: pin_name=%s sample_rate=%g number_of_samples=%d",
+        pin_name,
         sample_rate,
         number_of_samples,
     )
+    session_management_client = nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            session_management_client.reserve_sessions(
+                context=measurement_service.context.pin_map_context,
+                pin_or_relay_names=[pin_name],
+                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+                # If another measurement is using the session, wait for it to complete.
+                # Specify a timeout to aid in debugging missed unreserve calls.
+                # Long measurements may require a longer timeout.
+                timeout=60,
+            )
+        )
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+        session_info = reservation.session_info[0]
+
+        def cancel_callback(task: Optional[nidaqmx.Task] = None):
+            logging.info("Canceling measurement")
+            task_to_abort = task
+            if task_to_abort is not None:
+                task_to_abort.control(TaskMode.TASK_ABORT)
```

Comment:
This doesn't work, for reasons described here: https://github.com/ni/measurementlink-python/pull/268#discussion_r1190356890
 
Please test that the cancel callback works:
- Run NI IO Trace
- Run a DAQmx measurement in InstrumentStudio. Increase the number of samples so it takes longer.
- Before it's done, click the stop button in InstrumentStudio
- Verify that NI IO Trace displays the call to `DAQmxTaskControl(DAQmx_Val_Task_Abort)`

Reply:
Sure. After the suggested change in https://github.com/ni/measurementlink-python/pull/268#discussion_r1190356890,  I have tested with NI I/O Trace and able to trace the call to `DAQmxTaskControl(DAQmx_Val_Task_Abort)`

![image](https://github.com/ni/measurementlink-python/assets/104556074/976f8fb4-8623-4b4d-ba3f-d05f96da09b8)

Reply:
Increase the number of samples so it takes longer. For example, 10000 samples at 1000 S/s takes 10 seconds. This makes the effects of the stop button more apparent.

Anyway, it works:
![image](https://github.com/ni/measurementlink-python/assets/17914399/aa838156-1194-402f-8383-f54749d20294)